File: LeetCode/Mathematics/LC-1979 GCD of min and max of Array/LC1979.java
class LC1979 {
    public static int findGCD(int[] nums) {
        int min = Integer.MAX_VALUE;
        int max = 0;
        for(int i=0; i<nums.length; i++){
            if(nums[i] <= min)
                min = nums[i];
            if(nums[i] > max)
                max = nums[i];
        }

        for(int j = (min+max)/2; j>0; j--){
            if(min%j==0 && max%j==0)
                return j;
        }
        return 1;
    }
    
    public static void main(String[] args){
        System.out.println(findGCD(new int[]{2,5,6,9,10}));
    }
}

File: LeetCode/Mathematics/LC-1979 GCD of min and max of Array/LC1979.cs
using System;

class LC1979
{
    public static int FindGCD(int[] nums)
    {
        int min = int.MaxValue;
        int max = 0;

        // Find the minimum and maximum values in the array
        foreach (int num in nums)
        {
            if (num <= min)
                min = num;
            if (num > max)
                max = num;
        }

        // Calculate the GCD of the minimum and maximum
        for (int j = (min + max) / 2; j > 0; j--)
        {
            if (min % j == 0 && max % j == 0)
                return j;
        }

        return 1; // Default return value if no GCD is found
    }

    public static void Main(string[] args)
    {
        Console.WriteLine(FindGCD(new int[] { 2, 5, 6, 9, 10 })); // Output: 1
    }
}
